WordPressKit/HTTPClient.swift

```diff
@@ -0,0 +1,68 @@
+import Foundation
+
+public typealias WordPressAPIResult<R, E: LocalizedError> = Result<R, WordPressAPIError<E>>
+
+struct HTTPAPIResponse<Body> {
+    typealias Body = Body
+
+    var response: HTTPURLResponse
+    var body: Body
+}
+
+extension URLSession {
+
+    func apiResult<E: LocalizedError>(
+        with builder: HTTPRequestBuilder,
+        errorType: E.Type = E.self
```

Comment:
`errorType` looks unused?

<img width="793" alt="image" src="https://github.com/wordpress-mobile/WordPressKit-iOS/assets/1218433/285c8f45-cf30-444d-9569-a18255a53663">

Reply:
I'll keep rolling with the unused assumption and make an additional suggestion, which would be valid anyway but results in my opinion cleaner this way. 

What do you think of renaming the method to a verb like "perform"? The fact that the method returns an API result is already encoded in the type signature.

This way, including the removed error type, we could go from a usage like:

```swift
let result = await URLSession.shared.apiResult(with: .init(url: URL(string: "https://wordpress.org/hello")!), errorType: TestError.self)
```

to:

```swift
let result = await URLSession.shared.perform(.init(url: URL(string: "https://wordpress.org/hello")!)
```

or, if I misunderstood the error type requirement, 

```swift
let result = await URLSession.shared.perform(.init(url: URL(string: "https://wordpress.org/hello")!), errorType: TestError.self)
```

An alternative to `perform` could also be `request`

Reply:
> errorType looks unused?

Yes. But that's intended. We don't really need it in the implementation. It's similar to `JSONEncoder.encode` where callers can use it to tell Swift compiler what the `E` type is instead of letting the compiler infers it based on the context. You can see examples of this usage in the unit tests.

> What do you think of renaming the method to a verb like "perform"?

Sounds good to me! 👍

Reply:
> [`errorType` works] similar to JSONEncoder.encode where callers can use it to tell Swift compiler what the E type is instead of letting the compiler infers it based on the context.

Gotcha. If we don't specify it, then we're left to explicitly set the type of the result we expect from the API call, as can be seen in this edit I made locally.

![image](https://github.com/wordpress-mobile/WordPressKit-iOS/assets/1218433/38206b08-6fc5-4412-8394-e5688487be8f)

To be honest, I'd rather to that than pass unused arguments, but I can see how the current version is easier on the eyes. Also, as you point out, `JSONEncoder` and `JSONDecoder` set a precedent for this style, which might then result more familiar to Swift devs.